truckersmp_cli/__init__.py

```diff
@@ -718,59 +725,110 @@ def get_supported_game_versions():
+        except Exception as e:
+            sys.exit(
+                "Failed to convert game directory to Windows path: {}".format(e))
+
+        steamcmd = os.path.join(Dir.steamcmddir, "steamcmd.exe")
+        steamcmd_cmd.append(wine)
+        steamcmd_url = URL.steamcmdwin
+    steamcmd_cmd.append(steamcmd)
+
+    # fetch SteamCMD if not in our data directory
+    os.makedirs(Dir.steamcmddir, exist_ok=True)
     if not os.path.isfile(steamcmd):
-        logging.debug("Downloading SteamCMD")
-        os.makedirs(Dir.steamcmddir, exist_ok=True)
+        logging.debug("Retrieving SteamCMD")
         try:
-            with urllib.request.urlopen(URL.steamcmdurl) as f:
-                steamcmd_targz = f.read()
+            with urllib.request.urlopen(steamcmd_url) as f:
+                steamcmd_archive = f.read()
         except Exception as e:
-            sys.exit("Failed to download SteamCMD: {}".format(e))
-        with tarfile.open(fileobj=io.BytesIO(steamcmd_targz), mode="r:gz") as f:
-            f.extractall(Dir.steamcmddir)
+            sys.exit("Failed to retrieve SteamCMD: {}".format(e))
+        logging.debug("Extracting SteamCMD")
+        try:
+            if args.proton:
+                with tarfile.open(
+                  fileobj=io.BytesIO(steamcmd_archive), mode="r:gz") as f:
+                    f.extractall(Dir.steamcmddir)
+            else:
+                with ZipFile(io.BytesIO(steamcmd_archive)) as f:
+                    with f.open("steamcmd.exe") as f_exe:
+                        with open(steamcmd, "wb") as f_out:
+                            f_out.write(f_exe.read())
+        except Exception as e:
+            sys.exit("Failed to extract SteamCMD: {}".format(e))
+
     logging.info("SteamCMD: " + steamcmd)
 
-    # download/update Proton
     if args.proton:
-        logging.debug("Updating Proton (AppID:{})".format(args.proton_appid))
-
-        if not os.path.isdir(args.protondir):
-            logging.debug("Creating directory {}".format(args.protondir))
-            os.makedirs(args.protondir, exist_ok=True)
+        if check_steam_process(use_proton=True):
+            logging.debug("Closing Steam")
+            subproc.call(("steam", "-shutdown"))
 
+        # download/update Proton
+        os.makedirs(args.protondir, exist_ok=True)
+        logging.debug("Updating Proton (AppID:{})".format(args.proton_appid))
         logging.info("""Command:
   {}
     +login {}
     +force_install_dir {}
     +app_update {} validate
     +quit""".format(steamcmd, args.account, args.protondir, args.proton_appid))
         subproc.call(
-          [steamcmd,
+          (steamcmd,
            "+login", args.account,
            "+force_install_dir", args.protondir,
            "+app_update", str(args.proton_appid), "validate",
-           "+quit"])
+           "+quit"))
+    else:
+        if check_steam_process(use_proton=False, wine=wine, env=env_steam):
```

Comment:
The windows version of steamcmd also causes the native steam client to loose it's connection. We need to test both. Same for the native steamcmd above I guess.

Not sure what's the best way here. Maybe ditch the `use_proton=True/False` in `check_steam_process()` and also make it close them in case there is one found?

Reply:
Another possibility is to not care at the connection loss. I haven't seen Steam logging me out because of that recently, that part was the reason I closed steam in the shell script.